File: backend/src/main/java/edu/esprit/kaddem/dto/ProfesseurDto.java
package edu.esprit.kaddem.dto;

import edu.esprit.kaddem.model.Classe;
import edu.esprit.kaddem.model.user.Etudiant;
import edu.esprit.kaddem.model.user.Parent;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.List;

/**
 * A DTO for the {@link Etudiant} entity
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class ProfesseurDto extends AbstractUserDto<Parent> implements Serializable {
    private List<ClasseDto> classes;
    private List<MatiereDto> matieres;
}

File: frontend/src/gs-api/src/models/professeur-dto.ts
/* tslint:disable */
import { ClasseDto } from './classe-dto';
import { MatiereDto } from './matiere-dto';
export interface ProfesseurDto {
  classes?: Array<ClasseDto>;
  dateDeNaissance?: string;
  email?: string;
  id?: number;
  matieres?: Array<MatiereDto>;
  nom?: string;
  numTel?: string;
  prenom?: string;
}
